examples/AUTHENTICATION_EXAMPLE.md

```diff
@@ -0,0 +1,98 @@
+Authentication Example
+==============================================
+
+The example requires grpc-java to be already built. If you use a release tag
+the build will download the relevant binaries from a maven repository.
+Otherwise you must follow [COMPILING](../COMPILING.md) to build the latest
+SNAPSHOT binaries.
+
+The source is [here](src/main/java/io/grpc/examples/authentication).
+
+To build the example,
+
+1. **[Install gRPC Java library SNAPSHOT locally, including code generation plugin](../../COMPILING.md) (Only need this step for non-released versions, e.g. master HEAD).**
+
+2. Run in this directory:
+```
+$ ../gradlew installDist
+```
+
+This creates the scripts `auth-server` and `auth-client` in the
+`build/install/examples/bin/` directory that can be used to run this example.
+The example requires the server to be running before starting the client.
+
+Running the auth-server is the same as the normal hello world example i.e. there are no arguments to supply:
+
+**auth-server**:
+
+```text
+USAGE: AuthServer
+```
+
+
+**auth-client**:
+
+```text
+USAGE: AuthClient [user-name] [token-value]
+```
+
+#### Note
+
+The example shows how simple JWT-like credential based authentication is implemented in gRPC using
+client and server interceptors. For simplicity a simple string value is used instead of a string-encoded JWT.
+
+
+#### Auth example:
+
+```bash
+# Run the server:
+./build/install/example-tls/bin/hello-world-tls-server localhost 50440 /tmp/sslcert/server.crt /tmp/sslcert/server.pem
+# In another terminal run the client
+./build/install/example-tls/bin/hello-world-tls-client localhost 50440 /tmp/sslcert/ca.crt
+```
+
```

Comment:
I don't understand the tls thing here. How is it related to auth server and client?

Reply:
Oops, that's a mistake. Will fix.

Reply:
fixed